tests/aggregation_regress/tests.py

```diff
@@ -360,6 +361,75 @@ def test_aggrate_annotation(self):
             {'c__max': 3}
         )
 
+        # Regression for #25307:  AttributeError: 'WhereNode' object has no
+        # attribute 'get_source_expressions'. Conditional aggregation of a
+        # grouped queryset.
+        self.assertEqual(
+            Book.objects.annotate(c=Count('authors')).values('c').aggregate(test=Sum(
```

Comment:
This isn't passing on any backends. SQLite, for example: `OperationalError: misuse of aggregate function COUNT()`.